### PR TITLE
LocalFolder and improvements in logging and prediction pipelines

### PR DESCRIPTION
**Pull request by AleksMat**
All these changes were done for a use case of having a prediction pipeline that is using a Tensorflow model.

- Parameter `model_filename` was moved from `BasePredictionPipeline` to its 2 child classes. That is because a Tensorflow model consists of a folder structure with multiple files and it is nicer to handle its name a bit differently.
- Implemented `LocalFolder` utility, improved the overall implementation of `LocalFile`/`LocalFolder`, and added more unit tests.
- Improvements in logs:
  * Figured out how to explicitly prevent the crashes because of emitting logs within emitting logs - by adding a default filter `FilesystemFilter` to the `FilesystemHandler`.
  * Fixed minor issues with config parameters.

### CODE REVIEW

File: eogrow/utils/fs.py [hunk 1]
```diff
@@ -61,19 +65,19 @@ def __init__(
             self._local_path = fs.path.basename(self._remote_path)
 
         self._absolute_local_path = self._filesystem.getsyspath(self._local_path)
+        self._remote_location_ensured = False
```
Comment (zigaLuksic) [inline eogrow/utils/fs.py:68]: what is the usecase where this results in an improvement?

**Review by zigaLuksic — APPROVED** [review]

File: eogrow/utils/fs.py (hunk 1 above)
Comment (AleksMat) [reply eogrow/utils/fs.py:68]: Something like this:

```Python
with LocalFile("/path/to/file/something.json", mode="w") as local_file:
    # At this point we don't want that path to file is generated yet
    ...
    local_file.copy_to_remote()  # Now the path should be generated
    ...
    local_file.copy_to_remote()  # Path already exists so we don't even have to make IO calls to ensure it exists

...
# Once local_file is closed the file is copied again but again we don't have to ensure that the path exists
```

So basically it aims to reduce a few unnecessary IO calls to remote filesystem.

Comment (zigaLuksic) [reply eogrow/utils/fs.py:68]: but do we want the user to manually copy the file over multiple times?

Comment (AleksMat) [reply eogrow/utils/fs.py:68]: In some cases, yes. In our case, we use this for copying logs to a remote filesystem every minute or so. This way, logs are available for users to see right away and if at any time an EC2 instance dies the logs are already on S3.

Comment (zigaLuksic) [reply eogrow/utils/fs.py:68]: using it in logging is reason enough, good point

**Pull request merged**